src/realm/query_expression.hpp

```diff
@@ -2910,6 +2911,126 @@ class Columns<Lst<T>> : public Subexpr2<T>, public ColumnListBase {
     }
 };
 
+template <typename T>
+class Columns<Set<T>> : public Subexpr2<T>, public ColumnListBase {
```

Comment:
A common base class for `Columns<Lst<T>>` and `Columns<Set<T>>` should be made.

Reply:
I very much agree, but I considered it out of scope for this. I would rather make such a refactor in a separate PR, if that's OK with you.

Reply:
I think this is necessary to do now in order to make the aggregate functions work. I have dona that in additional commit.